packages/core/test/features/st-var.spec.ts

```diff
@@ -1259,162 +1284,569 @@ describe(`features/st-var`, () => {
+                });
             });
 
-            const { meta } = sheets['/entry.st.css'];
-            const computedVars = stylable.stVar.getComputed(meta);
-
-            expect(Object.keys(computedVars)).to.eql(['border']);
-            expect(computedVars.border).to.containSubset({
-                value: '1px solid red',
-                input: {
-                    color: 'red',
-                    size: '1px',
-                    style: 'solid',
-                },
-                diagnostics: { reports: [] },
+            it('should emit diagnostics only on invalid computed st-vars', () => {
+                const { stylable, sheets } = testStylableCore(
+                    `
+                    :vars {
+                        validBefore: red;
+                        invalid: invalid-func(imported);
+                        validAfter: green;
+                    }
+                    `
+                );
+
+                const { meta } = sheets['/entry.st.css'];
+
+                const computedVars = stylable.stVar.getComputed(meta);
+
+                expect(Object.keys(computedVars)).to.eql(['validBefore', 'invalid', 'validAfter']);
+                expect(computedVars.validBefore).to.containSubset({
+                    value: 'red',
+                    input: {
+                        flatValue: 'red',
+                        type: 'string',
+                        value: 'red',
+                    },
+                    diagnostics: { reports: [] },
+                });
+                expect(computedVars.validAfter).to.containSubset({
+                    value: 'green',
+                    input: {
+                        flatValue: 'green',
+                        type: 'string',
+                        value: 'green',
+                    },
+                    diagnostics: { reports: [] },
+                });
+                expect(computedVars.invalid).to.containSubset({
+                    value: 'invalid-func(imported)',
+                    input: {
+                        flatValue: 'invalid-func(imported)',
+                        type: 'string',
+                        value: 'invalid-func(imported)',
+                    },
+                    diagnostics: {
+                        reports: [
+                            {
+                                message: functionWarnings.UNKNOWN_FORMATTER('invalid-func'),
+                                type: 'warning',
+                            },
+                        ],
+                    },
+                });
+            });
+
+            it('should emit diagnostics only on invalid custom st-vars', () => {
+                const { stylable, sheets } = testStylableCore({
+                    '/entry.st.css': `
+                        @st-import [stBorder] from './st-border.js';
+    
+                        :vars {
+                            border: stBorder(st-array(1px, 2px), solid, red);
```

Comment:
can't we also add the diagnostic as part of the transformation resolve?

Reply:
Add a skipped test that will check this problem, it will be implemented in a new PR.